### PR TITLE
initial cookbook

### PR DESCRIPTION
**Pull request by armiller**
This is a varnish wrapper cookbook that defaults varnish to 3.0 and disables varnishlog while enabling varnishncsa.

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -2,4 +2,32 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Default to 3
+node.default['varnish']['version'] = 3.0
+
+# Disable varnishlog daemon to conserve disk space.
+node.default['varnish']['log_daemon'] = false
```
Comment (ramereth) [inline recipes/default.rb]: You should add a test to ensure this service is disabled and stopped.

**Comment by ramereth** [comment]
@armiller You might take a look at my comments and this pull request rackspace-cookbooks/varnish#44. It'd be nice to have settings for the sysconfig on ncsa logging.

Also considering this is a wrapper cookbook, I'm wondering if its worth it if we provide a default config that works out of the box with our default settings. We can obviously change this in other cookbooks easily, but it might be nice to have our default config at least here. Doesn't need to be crazy configurable. Thoughts?

File: .kitchen.cloud.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  openstack_auth_url: <%= "#{ENV['OS_AUTH_URL']}/tokens" %>
+  key_name: <%= ENV['OS_SSH_KEYPAIR'] %>
+  private_key_path: <%= ENV['OS_PRIVATE_SSH_KEY'] %>
+  public_key_path: <%= ENV['OS_PUBLIC_SSH_KEY'] %>
+  openstack_tenant: <%= ENV['OS_TENANT_NAME'] %>
+  floating_ip_pool: <%= ENV['OS_FLOATING_IP_POOL'] %>
+  flavor_ref: <%= ENV['OS_FLAVOR_REF'] %>
+  # Only use the following if you need to open up firewall ports
+  security_groups:
+    - <%= ENV['OS_SECURITY_GROUP_NO_FIREWALL'] %>
```
Comment (ramereth) [inline .kitchen.cloud.yml]: Remove this, its not needed for this.

File: recipes/default.rb [hunk 3]
```diff
@@ -2,4 +2,32 @@
 # Cookbook Name:: osl-varnish
 # Recipe:: default
 #
-# Copyright (c) 2015 The Authors, All Rights Reserved.
+# Copyright 2014, Oregon State University
```
Comment (ramereth) [inline recipes/default.rb]: Its now 2015 :)

File: test/integration/default/serverspec/server_spec.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require 'serverspec'
+
+set :backend, :exec
+
+describe service('varnishlog') do
```
Comment (ramereth) [inline test/integration/default/serverspec/server_spec.rb:5]: Its probably a good idea to add a few simple tests for varnish itself. Make sure the service is running with the correct port, etc. Also check the admin port is there. Do we want to include the firewall recipe for http here too? I think there's a varnish recipe too?

Comment (armiller) [inline test/integration/default/serverspec/server_spec.rb:5]: I don't see the point in adding tests for the actual cookbook. I would rather open a pr on the actual cookbook and add tests.

I can add the firewall recipes.

**Comment by armiller** [comment]
@ramereth so I can add a default varnish config file but it could cause confusion since I would be overriding the `node['varnish']['default_vcl']` attribute in the wrapper cookbook. If you got the order for the varnish attributes wrong you might accidentally install the default vcl instead of a custom one. Thoughts?

**Comment by armiller** [comment]
@ramereth what exactly do you want to manage with `varnishncsa`?

File: test/integration/default/serverspec/server_spec.rb (hunk 4 above)
Comment (ramereth) [inline test/integration/default/serverspec/server_spec.rb:5]: I don't mean adding a full test suite for varnish, just adding two simple checks that specific ports are listening that should be and the service is actually running. That's it.

**Comment by ramereth** [comment]
@armiller Let's hold off on messing with a default config in another PR.

@armiller WRT varnishncsa, I'm talking about managing `/etc/sysconfig/varnishncsa`. Look on www3.phpbb to see what I mean. By default it doesn't get installed, but it gets sourced by the init script if it's there. It should pull in the default config by default which is already set in the init script. But we need to change that sometimes on our end.

**Comment by armiller** [comment]
@ramereth ping for review

**Comment by armiller** [comment]
@jordane I remember talking about managing the varnish secret. Should that be included in this pr? Would that be managed by a global data bag?

**Comment by armiller** [comment]
@ramereth @jordane reping. This pr seems to have been lost.

**Comment by ramereth** [comment]
I think this looks LTGM. We can do other features later but this gets the initial stuff setup. :+1:

**Comment by ramereth** [comment]
Don't forget to add this to the mr config too.

**Comment by jordane** [comment]
+1

**Pull request merged**